src/main/java/de/ait/secondlife/controllers/ImageContorller.java

```diff
@@ -0,0 +1,131 @@
+                mediaType = "application/json",
+                schema = @Schema(implementation = ResponseMessageDto.class)
+        )),
+        @ApiResponse(responseCode = "500", description = "Internal server error", content = @Content(
+                mediaType = "application/json",
+                schema = @Schema(implementation = ResponseMessageDto.class)
+        ))
+})
+public class ImageContorller {
+
+    private final ImageService imageService;
+
+    @PostMapping("/upload")
+    @Operation(
+            summary = "Create new image",
+            description = "Create new image for entity by entity id "
+    )
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Successful operation",
+                    content = @Content(mediaType = "application/json",
+                            schema = @Schema(implementation = ResponseMessageDto.class))),})
+    public ResponseEntity<ResponseMessageDto> uploadImage(
+            @Valid
+            @Parameter(description = "Dto with image file, entity type and entity id ", schema = @Schema(implementation = ImageCreateDto.class))
+            ImageCreateDto request) {
+        imageService.saveNewImage(request);
+        return ResponseEntity.ok(
+                new ResponseMessageDto("Image(s) successful saved"));
+    }
+
+    @GetMapping
+    @Operation(
+            summary = "Get names of entity images",
+            description = "Get all names of image of entity by type and id"
+    )
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Successful operation",
+                    content = @Content(mediaType = "application/json"
+                            , schema = @Schema(implementation = ImagePathsResponseDto.class)
+                    ))})
+    public ResponseEntity<ImagePathsResponseDto> getImages(
+            @RequestParam
+            @Parameter(description = "Code of type of entity", examples = {
+                    @ExampleObject(name = "offer", value = "1"),
+                    @ExampleObject(name = "user", value = "2"),
+                    @ExampleObject(name = "category", value = "3")
+            })
+            int typeCode,
+            @RequestParam
+            @Parameter(description = "Id of entity", example = "343")
+            Long id) {
+        String entityType = EntityType.get(typeCode).getType();
+        return ResponseEntity.ok(imageService.findAllImageForEntity(entityType, id));
+    }
+
+  @PostMapping
+  @Operation(
+          summary = "Get image",
+          description = "Get image by filename"
+  )
+  @ApiResponses(value = {
+          @ApiResponse(responseCode = "200", description = "Successful operation",
+                  content = @Content(mediaType = "application/json"
+                          , schema = @Schema(implementation = StreamingResponseBody.class)
+                  ))})
+    public ResponseEntity<StreamingResponseBody> getImageByFile(
```

Comment:
at the moment I don’t understand where this endpoint can be used. worth discussing

Reply:
endpoint removed